Add NVMe-oF HII feature to timberland_upstream

trevor-cockrell opened the pull request:
3 commits introduced, divided by module:

- NetworkPkg
  - Addition of PcdMaxNvmeOfAttemptNumber
- NetworkPkg/Include
  - Addition of gNvmeOfConfigGuid
- NetworkPkg/NvmeOfDxe
  - Add HII/configuration forms
  - Add HII details in NvmeOfDxe README

trevor-cockrell committed:
NetworkPkg: Add Pcd for maximum NVMe-oF attempts

Add PcdMaxNvmeOfAttemptNumber to NetworkPkg as the maximum number of
NVMe-over-fabrics attempts.

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

trevor-cockrell committed:
NetworkPkg/Include: Add GUID gNvmeOfConfigGuid

GUID gNvmeOfConfigGuid added for NvmeOfDxe HII configuration forms.

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

trevor-cockrell committed:
NetworkPkg/NvmeOfDxe: NVMe-over-Fabrics configuration forms

Add support for modification of NVMe-oF connection attempts in the
NVMe-oF configuration submenu of the HII.
Update NvmeOfDxe readme with configuration form details.

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

CurtisBallard commented:
Accidentally clicked "Ready for review" and changed state without intending to.  Reverted back to draft.

trevor-cockrell committed:
NetworkPkg/NvmeOfDxe: Convert Host NID to Host ID

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

trevor-cockrell committed:
NetworkPkg/NvmeOfDxe: Make host gateway optional

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

trevor-cockrell committed:
NetworkPkg/NvmeOfDxe: Populate Host NQN/ID according to TP-4126

During global data initialization: if empty, the Host NQN and/or
Host ID will be populated based on the system UUID.

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

trevor-cockrell committed:
NetworkPkg/NvmeOfDxe: Various fixes

  - Resolve MSVCC compilation errors with VS2019
  - NvmeOfConfig: improve checking for empty .v6.Addr values when
    converting NV data to IFR data

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

trevor-cockrell committed:
NetworkPkg/NvmeOfDxe: NvmeOfPassthru HII compatability

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

trevor-cockrell commented:
Closing as https://github.com/timberland-sig/edk2/pull/25 integrates these changes + fixes from https://github.com/timberland-sig/edk2/pull/24.
